src/services/AppUserService.ts

```diff
@@ -18,4 +32,34 @@ export class AppUserService {
   getMultipleAppUsers(ids: number[]): Promise<AppUser[]> {
     return this.appUserRepository.find({ id: Any(ids) });
   }
+
+  /**
+   * Get an existing AppUser by their email address.
+   *
+   * @param email The email used to look for the corresponding AppUser.
+   *
+   */
+  getAppUserByEmail(email: string): Promise<AppUser> {
+    return this.appUserRepository.findOne({ email });
+  }
+
+  async saveNonAffiliate(appUserRequest: EventSignInRequest): Promise<AppUser | undefined> {
```

Comment:
hm I think it would be nice if services didn't need to know about payloads. what if we can just have a requestToAppUser in the mapper? then saveNonAffiliate can just take in AppUser instead of AppUserRequest

Reply:
Yeah I can do that. We already have requestToNewEntity which is basically requestToAppUser since it just creates a new AppUser entity (with no saves).